### PR TITLE
Fix the Jacobian calculation for spherically symmetric, 1D elements

### LINKED ISSUE #19570: Jacobian for tensor_mechanics spherical
## Bug Description
The Jacobian calculation in StressDivergenceRSphericalTensors.C is incorrect.

## Steps to Reproduce
Running a simple elasticity problem with Newton as the solve type shows this.

## Impact
Fixing this will potentially improve the calculation times for spherically symmetric thermomechanical problems.

### PR DESCRIPTION
**Pull request by jasondhales**
Also simplified the finite strain calculation for RSpherical.

Closes #19570

### CODE REVIEW

File: modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C [hunk 1]
```diff
@@ -66,27 +66,34 @@ StressDivergenceRSphericalTensors::computeQpOffDiagJacobian(unsigned int jvar)
 }
 
 Real
+#if DEBUG
 StressDivergenceRSphericalTensors::calculateJacobian(unsigned int ivar, unsigned int jvar)
+#else
+StressDivergenceRSphericalTensors::calculateJacobian(unsigned int /*ivar*/, unsigned int /*jvar*/)
+#endif
 {
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C:74]:
```suggestion
StressDivergenceRSphericalTensors::calculateJacobian(unsigned int ivar, unsigned int jvar)
{
  libmesh_ignore(ivar);
  libmesh_ignore(jvar);
```

Comment (loganharbour) [reply modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C:74]: Even better:

```
StressDivergenceRSphericalTensors::calculateJacobian(unsigned int libmesh_dbg_var(ivar), unsigned int libmesh_dbg_var(jvar))
```

Comment (dschwen) [reply modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C:74]: Yeah, that's the one I had in mind but could not find anymore!

File: modules/tensor_mechanics/src/materials/ComputeRSphericalFiniteStrain.C [hunk 2]
```diff
@@ -87,3 +87,13 @@ ComputeRSphericalFiniteStrain::computeProperties()
...
+
+void
+ComputeRSphericalFiniteStrain::computeQpIncrements(RankTwoTensor & total_strain_increment,
+                                                   RankTwoTensor & rotation_increment)
+{
+  for (unsigned int i = 0; i < 3; ++i)
+    total_strain_increment(i, i) = std::log(_Fhat[_qp](i, i));
+
+  rotation_increment.setToIdentity();
+}
```
Comment (jiangwen84) [inline modules/tensor_mechanics/src/materials/ComputeRSphericalFiniteStrain.C:99]: shall we add this function to `ADComputeRSphericalFiniteStrain` too?

File: modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C [hunk 3]
```diff
@@ -66,27 +66,34 @@ StressDivergenceRSphericalTensors::computeQpOffDiagJacobian(unsigned int jvar)
...
+  const Real phi_r1 = _phi[_j][_qp] / _q_point[_qp](0);
+  const Real phi_r2 = phi_r1;
+
+  const Real term1 = test * _Jacobian_mult[_qp](0, 0, 0, 0) * phi;
+  const Real term2 = test * _Jacobian_mult[_qp](0, 0, 1, 1) * phi_r1; // same as term3
+  // const Real term3 = test * _Jacobian_mult[_qp](0, 0, 2, 2) * phi_r2;
+
+  const Real term4 = test_r1 * _Jacobian_mult[_qp](1, 1, 0, 0) * phi;    // same as term7
+  const Real term5 = test_r1 * _Jacobian_mult[_qp](1, 1, 1, 1) * phi_r1; // same as term9
+  const Real term6 = test_r1 * _Jacobian_mult[_qp](1, 1, 2, 2) * phi_r2; // same as term8
```
Comment (jiangwen84) [inline modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C:92]: does this work with orthotropic and/or anisotropic cases?

File: modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C [hunk 4]
```diff
@@ -66,27 +66,34 @@ StressDivergenceRSphericalTensors::computeQpOffDiagJacobian(unsigned int jvar)
...
+
+  // const Real term7 = test_r2 * _Jacobian_mult[_qp](2, 2, 0, 0) * phi;
+  // const Real term8 = test_r2 * _Jacobian_mult[_qp](2, 2, 1, 1) * phi_r1;
```
[the remark this answers is not in the document]
Comment (jasondhales) [reply modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C:95]: There just for info, but they are not zero.  term6 and term8 evaluate to the same thing.

File: modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C (hunk 3 above)
Comment (jasondhales) [reply modules/tensor_mechanics/src/kernels/StressDivergenceRSphericalTensors.C:92]: Good question.  It should, but without a test I can't say that I haven't messed up the i,j,k,l.  I'll add a test.

File: modules/tensor_mechanics/src/materials/ComputeRSphericalFiniteStrain.C (hunk 2 above)
Comment (jasondhales) [reply modules/tensor_mechanics/src/materials/ComputeRSphericalFiniteStrain.C:99]: I've added that to my latest changes.

**Comment by hugary1995** [comment]
The `ave_Fhat` isn't being used anywhere.
https://github.com/idaholab/moose/blob/187befbe8da08ee71cb2ffa31141ee80bd97f2db/modules/tensor_mechanics/src/materials/ComputeRSphericalFiniteStrain.C#L48

This isn't directly related to this PR, but maybe you could fix this while you are at it :)

**Comment by jasondhales** [comment]
Good point, @hugary1995.  I've removed that line.

**Review by jiangwen84 — APPROVED** [review]

**Comment by jiangwen84** [comment]
A Bison PR is ready to fix the failure.

**Pull request merged**